### PR TITLE
Minor changes to the pull request checklist

### PR DESCRIPTION
**Pull request by agbutteriss**
I've added a line about including a what's new entry (see #2104), and removed the stuff about iris-code-generators at @pp-mo's suggestion, since it's obsolete.

### CODE REVIEW

**Comment by agbutteriss** [comment]
Now I'm looking at it, it seems like it could be worth including "have you added any dependencies to conda-requirements.txt/the install page" as well.

**Comment by pp-mo** [comment]
> @agbutteriss Now I'm looking at it, it seems like it could be worth including "have you added any dependencies to conda-requirements.txt/the install page" as well. 

Definitely worth it :+1:

File: docs/iris/src/developers_guide/pulls.rst [hunk 1]
```diff
@@ -62,20 +63,20 @@ The Iris Check List
 
  * ``make doctest``, ``make extest``  in ``./docs/iris``.
```
Comment (agbutteriss) [inline docs/iris/src/developers_guide/pulls.rst]: It seems like ``make extest`` isn't actually a thing, but before I remove it does anyone know what, if anything, replaces it and if it's worth including here?

Comment (ajdawson) [inline docs/iris/src/developers_guide/pulls.rst]: `extest` is a target in the documentation Makefile that runs tests on the example code (code that produces gallery examples).

Comment (agbutteriss) [inline docs/iris/src/developers_guide/pulls.rst]: Ah, I was in iris/docs/src which has its own Makefile. Thanks.

**Pull request merged**